src/main/java/dal/fatih/todoly/Main.java

```diff
@@ -30,10 +33,14 @@ public static Task handleCreateTask() {
                 System.out.println("The given date can not be older than now");
 
             } else {
+
                 UUID uniqId = UUID.randomUUID();
                 Task task = new Task(uniqId, title, description, dueDate);
                 tasks.put(uniqId.toString(), task);
                 System.out.println(task.getTitle() + " titled task added");
+                outputTask = new ObjectOutputStream(new FileOutputStream(file));
+                outputTask.writeObject(tasks);

```

Comment:
Create a function out of this and invoke that instead of doing it multiple times. You do the same thing in `delete`